client/src/app/hooks/table-controls/useSelectionState.ts

```diff
@@ -0,0 +1,92 @@
+import { useState, useCallback, useMemo } from "react";
+
+export interface ISelectionStateArgs<T> {
```

Comment:
Please go first with functional changes (and keep them minimal) -  after moving and re-formatting it's hard to review what's changed i.e.  `isItemSelectable` is gone. It was not very useful in our app but still this change should be easy to notice and acknowledge

Reply:
Two things were pulled out `isItemSelectable` (as mentioned in the PR description) and the `externalState`.  Neither were used and the dead code was removed.

The rest of the changes swapped the change functions out to better match up to the needs of UI components.

Reply:
And yeah -- git decided that I changed the file enough that it wasn't a rename and update, it was a remove and this is a new file.  I really don't want to just rearrange the code to make the diff smaller and just be a rename plus change.

Reply:
Rearranged things, diffs should look better now.  New comments in a new thread.